x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -125,7 +126,8 @@ protected static Batch<LogicalPlan> substitutions() {
             new ReplaceRegexMatch(),
             new ReplaceAliasingEvalWithProject(),
             new SkipQueryOnEmptyMappings(),
-            new SubstituteSpatialSurrogates()
+            new SubstituteSpatialSurrogates(),
+            new ReplaceOrderByExpressionWithEval()
```

Comment:
Thought: There's 4 places now that take expressions out of a plan node and turn them into evals:
- `ReplaceStatsNestedExpressionWithEval` and this create an `Eval` before the node.
- `ReplaceStatsAggExpressionWithEval` and `SubstituteSurrogates` create an `Eval` after the node.

I wonder if this should be consolidated into 2 opt. rules, or at least should use common helper methods.

It's probably fine for now, though.

Reply:
I agree with you in that we should look into the possibility of refactoring this.  But this should probably be done separately under the umbrella of "refactoring" or maybe "tech debt".